Docker-Compose demo broken after recent changes

**Describe the bug**
docker-compose demo does not work

**Error message**
```haystack-api_1   | [2021-09-14 06:26:06 +0000] [1] [INFO] Starting gunicorn 20.1.0
haystack-api_1   | [2021-09-14 06:26:06 +0000] [1] [INFO] Listening at: http://0.0.0.0:8000 (1)
haystack-api_1   | [2021-09-14 06:26:06 +0000] [1] [INFO] Using worker: uvicorn.workers.UvicornWorker
haystack-api_1   | [2021-09-14 06:26:06 +0000] [10] [INFO] Booting worker with pid: 10
haystack-api_1   | /usr/local/lib/python3.7/site-packages/ray/autoscaler/_private/cli_logger.py:61: FutureWarning: Not all Ray CLI dependencies were found. In Ray 1.4+, the Ray CLI, autoscaler, and dashboard will only be usable via `pip install 'ray[default]'`. Please update your install command.
haystack-api_1   |   "update your install command.", FutureWarning)
haystack-api_1   | pdftotext version 4.03 [www.xpdfreader.com]
haystack-api_1   | Copyright 1996-2021 Glyph & Cog, LLC
haystack-api_1   | 09/14/2021 06:26:08 - WARNING - elasticsearch -   HEAD http://localhost:9200/ [status:N/A request:0.001s]
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
haystack-api_1   |     (self._dns_host, self.port), self.timeout, **extra_kw
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
haystack-api_1   |     raise err
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
haystack-api_1   |     sock.connect(sa)
haystack-api_1   | ConnectionRefusedError: [Errno 111] Connection refused
haystack-api_1   | 
haystack-api_1   | During handling of the above exception, another exception occurred:
haystack-api_1   | 
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/elasticsearch/connection/http_urllib3.py", line 246, in perform_request
haystack-api_1   |     method, url, body, retries=Retry(False), headers=request_headers, **kw
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
haystack-api_1   |     method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 507, in increment
haystack-api_1   |     raise six.reraise(type(error), error, _stacktrace)
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/packages/six.py", line 770, in reraise
haystack-api_1   |     raise value
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
haystack-api_1   |     chunked=chunked,
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
haystack-api_1   |     conn.request(method, url, **httplib_request_kw)
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
haystack-api_1   |     super(HTTPConnection, self).request(method, url, body=body, headers=headers)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1244, in request
haystack-api_1   |     self._send_request(method, url, body, headers, encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1290, in _send_request
haystack-api_1   |     self.endheaders(body, encode_chunked=encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1239, in endheaders
haystack-api_1   |     self._send_output(message_body, encode_chunked=encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1026, in _send_output
haystack-api_1   |     self.send(msg)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 966, in send
haystack-api_1   |     self.connect()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
haystack-api_1   |     conn = self._new_conn()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
haystack-api_1   |     self, "Failed to establish a new connection: %s" % e
haystack-api_1   | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f5c04290790>: Failed to establish a new connection: [Errno 111] Connection refused
haystack-api_1   | 09/14/2021 06:26:08 - WARNING - elasticsearch -   HEAD http://localhost:9200/ [status:N/A request:0.000s]
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
haystack-api_1   |     (self._dns_host, self.port), self.timeout, **extra_kw
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
haystack-api_1   |     raise err
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
haystack-api_1   |     sock.connect(sa)
haystack-api_1   | ConnectionRefusedError: [Errno 111] Connection refused
haystack-api_1   | 
haystack-api_1   | During handling of the above exception, another exception occurred:
haystack-api_1   | 
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/elasticsearch/connection/http_urllib3.py", line 246, in perform_request
haystack-api_1   |     method, url, body, retries=Retry(False), headers=request_headers, **kw
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
haystack-api_1   |     method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 507, in increment
haystack-api_1   |     raise six.reraise(type(error), error, _stacktrace)
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/packages/six.py", line 770, in reraise
haystack-api_1   |     raise value
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
haystack-api_1   |     chunked=chunked,
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
haystack-api_1   |     conn.request(method, url, **httplib_request_kw)
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
haystack-api_1   |     super(HTTPConnection, self).request(method, url, body=body, headers=headers)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1244, in request
haystack-api_1   |     self._send_request(method, url, body, headers, encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1290, in _send_request
haystack-api_1   |     self.endheaders(body, encode_chunked=encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1239, in endheaders
haystack-api_1   |     self._send_output(message_body, encode_chunked=encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1026, in _send_output
haystack-api_1   |     self.send(msg)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 966, in send
haystack-api_1   |     self.connect()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
haystack-api_1   |     conn = self._new_conn()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
haystack-api_1   |     self, "Failed to establish a new connection: %s" % e
haystack-api_1   | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f5c042905d0>: Failed to establish a new connection: [Errno 111] Connection refused
haystack-api_1   | 09/14/2021 06:26:08 - WARNING - elasticsearch -   HEAD http://localhost:9200/ [status:N/A request:0.000s]
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
haystack-api_1   |     (self._dns_host, self.port), self.timeout, **extra_kw
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
haystack-api_1   |     raise err
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
haystack-api_1   |     sock.connect(sa)
haystack-api_1   | ConnectionRefusedError: [Errno 111] Connection refused
haystack-api_1   | 
haystack-api_1   | During handling of the above exception, another exception occurred:
haystack-api_1   | 
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/elasticsearch/connection/http_urllib3.py", line 246, in perform_request
haystack-api_1   |     method, url, body, retries=Retry(False), headers=request_headers, **kw
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
haystack-api_1   |     method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 507, in increment
haystack-api_1   |     raise six.reraise(type(error), error, _stacktrace)
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/packages/six.py", line 770, in reraise
haystack-api_1   |     raise value
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
haystack-api_1   |     chunked=chunked,
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
haystack-api_1   |     conn.request(method, url, **httplib_request_kw)
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
haystack-api_1   |     super(HTTPConnection, self).request(method, url, body=body, headers=headers)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1244, in request
haystack-api_1   |     self._send_request(method, url, body, headers, encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1290, in _send_request
haystack-api_1   |     self.endheaders(body, encode_chunked=encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1239, in endheaders
haystack-api_1   |     self._send_output(message_body, encode_chunked=encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1026, in _send_output
haystack-api_1   |     self.send(msg)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 966, in send
haystack-api_1   |     self.connect()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
haystack-api_1   |     conn = self._new_conn()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
haystack-api_1   |     self, "Failed to establish a new connection: %s" % e
haystack-api_1   | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f5d16d7c590>: Failed to establish a new connection: [Errno 111] Connection refused
haystack-api_1   | 09/14/2021 06:26:08 - WARNING - elasticsearch -   HEAD http://localhost:9200/ [status:N/A request:0.000s]
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 170, in _new_conn
haystack-api_1   |     (self._dns_host, self.port), self.timeout, **extra_kw
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 96, in create_connection
haystack-api_1   |     raise err
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/connection.py", line 86, in create_connection
haystack-api_1   |     sock.connect(sa)
haystack-api_1   | ConnectionRefusedError: [Errno 111] Connection refused
haystack-api_1   | 
haystack-api_1   | During handling of the above exception, another exception occurred:
haystack-api_1   | 
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/elasticsearch/connection/http_urllib3.py", line 246, in perform_request
haystack-api_1   |     method, url, body, retries=Retry(False), headers=request_headers, **kw
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 756, in urlopen
haystack-api_1   |     method, url, error=e, _pool=self, _stacktrace=sys.exc_info()[2]
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 507, in increment
haystack-api_1   |     raise six.reraise(type(error), error, _stacktrace)
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/packages/six.py", line 770, in reraise
haystack-api_1   |     raise value
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 706, in urlopen
haystack-api_1   |     chunked=chunked,
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 394, in _make_request
haystack-api_1   |     conn.request(method, url, **httplib_request_kw)
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 234, in request
haystack-api_1   |     super(HTTPConnection, self).request(method, url, body=body, headers=headers)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1244, in request
haystack-api_1   |     self._send_request(method, url, body, headers, encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1290, in _send_request
haystack-api_1   |     self.endheaders(body, encode_chunked=encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1239, in endheaders
haystack-api_1   |     self._send_output(message_body, encode_chunked=encode_chunked)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 1026, in _send_output
haystack-api_1   |     self.send(msg)
haystack-api_1   |   File "/usr/local/lib/python3.7/http/client.py", line 966, in send
haystack-api_1   |     self.connect()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 200, in connect
haystack-api_1   |     conn = self._new_conn()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/urllib3/connection.py", line 182, in _new_conn
haystack-api_1   |     self, "Failed to establish a new connection: %s" % e
haystack-api_1   | urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f5d16d7c9d0>: Failed to establish a new connection: [Errno 111] Connection refused
haystack-api_1   | [2021-09-14 06:26:08 +0000] [10] [ERROR] Exception in worker process
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/home/user/haystack/document_store/elasticsearch.py", line 202, in _init_elastic_client
haystack-api_1   |     f"Initial connection to Elasticsearch failed. Make sure you run an Elasticsearch instance "
haystack-api_1   | ConnectionError: Initial connection to Elasticsearch failed. Make sure you run an Elasticsearch instance at `[{'host': 'localhost', 'port': 9200}]` and that it has finished the initial ramp up (can take > 30s).
haystack-api_1   | 
haystack-api_1   | During handling of the above exception, another exception occurred:
haystack-api_1   | 
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/home/user/haystack/pipeline.py", line 431, in _load_or_get_component
haystack-api_1   |     instance = BaseComponent.load_from_args(component_type=component_type, **component_params)
haystack-api_1   |   File "/home/user/haystack/schema.py", line 285, in load_from_args
haystack-api_1   |     instance = subclass(**kwargs)
haystack-api_1   |   File "/home/user/haystack/document_store/elasticsearch.py", line 117, in __init__
haystack-api_1   |     ca_certs=ca_certs, verify_certs=verify_certs,timeout=timeout)
haystack-api_1   |   File "/home/user/haystack/document_store/elasticsearch.py", line 207, in _init_elastic_client
haystack-api_1   |     f"Initial connection to Elasticsearch failed. Make sure you run an Elasticsearch instance at `{hosts}` and that it has finished the initial ramp up (can take > 30s).")
haystack-api_1   | ConnectionError: Initial connection to Elasticsearch failed. Make sure you run an Elasticsearch instance at `[{'host': 'localhost', 'port': 9200}]` and that it has finished the initial ramp up (can take > 30s).
haystack-api_1   | 
haystack-api_1   | During handling of the above exception, another exception occurred:
haystack-api_1   | 
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/home/user/haystack/pipeline.py", line 428, in _load_or_get_component
haystack-api_1   |     cls._load_or_get_component(name=value, definitions=definitions, components=components)
haystack-api_1   |   File "/home/user/haystack/pipeline.py", line 434, in _load_or_get_component
haystack-api_1   |     raise Exception(f"Failed loading pipeline component '{name}': {e}")
haystack-api_1   | Exception: Failed loading pipeline component 'DocumentStore': Initial connection to Elasticsearch failed. Make sure you run an Elasticsearch instance at `[{'host': 'localhost', 'port': 9200}]` and that it has finished the initial ramp up (can take > 30s).
haystack-api_1   | 
haystack-api_1   | During handling of the above exception, another exception occurred:
haystack-api_1   | 
haystack-api_1   | Traceback (most recent call last):
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/gunicorn/arbiter.py", line 589, in spawn_worker
haystack-api_1   |     worker.init_process()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/uvicorn/workers.py", line 66, in init_process
haystack-api_1   |     super(UvicornWorker, self).init_process()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 134, in init_process
haystack-api_1   |     self.load_wsgi()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/gunicorn/workers/base.py", line 146, in load_wsgi
haystack-api_1   |     self.wsgi = self.app.wsgi()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
haystack-api_1   |     self.callable = self.load()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 58, in load
haystack-api_1   |     return self.load_wsgiapp()
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/gunicorn/app/wsgiapp.py", line 48, in load_wsgiapp
haystack-api_1   |     return util.import_app(self.app_uri)
haystack-api_1   |   File "/usr/local/lib/python3.7/site-packages/gunicorn/util.py", line 359, in import_app
haystack-api_1   |     mod = importlib.import_module(module)
haystack-api_1   |   File "/usr/local/lib/python3.7/importlib/__init__.py", line 127, in import_module
haystack-api_1   |     return _bootstrap._gcd_import(name[level:], package, level)
haystack-api_1   |   File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
haystack-api_1   |   File "<frozen importlib._bootstrap>", line 983, in _find_and_load
haystack-api_1   |   File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
haystack-api_1   |   File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
haystack-api_1   |   File "<frozen importlib._bootstrap_external>", line 728, in exec_module
haystack-api_1   |   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
haystack-api_1   |   File "/home/user/rest_api/application.py", line 8, in <module>
haystack-api_1   |     from rest_api.controller.router import router as api_router
haystack-api_1   |   File "/home/user/rest_api/controller/router.py", line 3, in <module>
haystack-api_1   |     from rest_api.controller import file_upload, search, feedback
haystack-api_1   |   File "/home/user/rest_api/controller/file_upload.py", line 34, in <module>
haystack-api_1   |     INDEXING_PIPELINE = Pipeline.load_from_yaml(Path(PIPELINE_YAML_PATH), pipeline_name=INDEXING_PIPELINE_NAME)
haystack-api_1   |   File "/home/user/haystack/pipeline.py", line 401, in load_from_yaml
haystack-api_1   |     component = cls._load_or_get_component(name=name, definitions=definitions, components=components)
haystack-api_1   |   File "/home/user/haystack/pipeline.py", line 434, in _load_or_get_component
haystack-api_1   |     raise Exception(f"Failed loading pipeline component '{name}': {e}")
haystack-api_1   | Exception: Failed loading pipeline component 'Retriever': Failed loading pipeline component 'DocumentStore': Initial connection to Elasticsearch failed. Make sure you run an Elasticsearch instance at `[{'host': 'localhost', 'port': 9200}]` and that it has finished the initial ramp up (can take > 30s).
haystack-api_1   | [2021-09-14 06:26:08 +0000] [10] [INFO] Worker exiting (pid: 10)
haystack-api_1   | [2021-09-14 06:26:09 +0000] [1] [INFO] Shutting down: Master
haystack-api_1   | [2021-09-14 06:26:09 +0000] [1] [INFO] Reason: Worker failed to boot.
haystack_haystack-api_1 exited with code 3
```

**Additional context**
the demo used to work fine before the commit #1430  

**To Reproduce**
clone the repo and run "docker-compose up --build"

--- a/docker-compose-gpu.yml
+++ b/docker-compose-gpu.yml
@@ -20,7 +20,7 @@ services:
       - 8000:8000
     environment:
       # See rest_api/pipelines.yaml for configurations of Search & Indexing Pipeline.
-      - ELASTICSEARCHDOCUMENTSTORE_PARAMS_HOST=elasticsearch
+      - DOCUMENTSTORE_PARAMS_HOST=elasticsearch
     restart: always
     depends_on:
       - elasticsearch

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -12,7 +12,7 @@ services:
       - 8000:8000
     environment:
       # See rest_api/pipelines.yaml for configurations of Search & Indexing Pipeline.
-      - ELASTICSEARCHDOCUMENTSTORE_PARAMS_HOST=elasticsearch
+      - DOCUMENTSTORE_PARAMS_HOST=elasticsearch
     restart: always
     depends_on:
       - elasticsearch

